src/plone/restapi/services/aliases/get.py

```diff
@@ -22,40 +31,60 @@ def __init__(self, context, request):
         self.context = context
         self.request = request
 
-    def reply_item(self):
+    def reply(self, query, manual, start, end):
         storage = getUtility(IRedirectionStorage)
+        portal_path = "/".join(self.context.getPhysicalPath()[:2])
         context_path = "/".join(self.context.getPhysicalPath())
-        redirects = storage.redirects(context_path)
-        aliases = [deroot_path(alias) for alias in redirects]
-        self.request.response.setStatus(200)
-        self.request.response.setHeader("Content-Type", "application/json")
-        return [{"path": alias} for alias in aliases], len(aliases)
-
-    def reply_root(self):
-        """
-        redirect-to - target
-        path        - path
-        redirect    - full path with root
-        """
-        batch = RedirectsControlPanel(self.context, self.request).redirects()
-        redirects = [entry for entry in batch]
 
-        for redirect in redirects:
-            del redirect["redirect"]
-            redirect["datetime"] = datetimelike_to_iso(redirect["datetime"])
-        self.request.response.setStatus(200)
+        if not IPloneSiteRoot.providedBy(self.context):
+            tree = OOBTree()
+            rds = storage.redirects(context_path)
+            for rd in rds:
+                rd_full = storage.get_full(rd)
+                tree[rd] = rd_full
+            storage = tree
+        else:
+            storage = storage._paths
+
+        if query and query.startswith("/"):
+            min_k = f"{portal_path}/{query.strip('/')}"
+            max_k = min_k[:-1] + chr(ord(min_k[-1]) + 1)
+            redirects = storage.items(min=min_k, max=max_k, excludemax=True)
+        elif query:
+            redirects = [path for path in storage.items() if query in path]
+        else:
+            redirects = storage.items()
 
-        self.request.form["b_start"] = "0"
-        self.request.form["b_size"] = "1000000"
-        self.request.__annotations__.pop("plone.memoize")
+        aliases = []
+        for redirect in redirects:
```

Comment:
I would write this as 
```suggestion
        for path, info in redirects:
```

That way Python destructures the items for you so you don't have to do `info = redirect[1]` and `redirect[0]` below. It's just a bit easier to read.